File: searching/python/binarySearch.py
import math

testArray = [1, 3, 4, 5, 9, 11, 18, 22, 31, 42, 69, 100, 101, 102, 450, 999];

def binarySearch(array, searchFor):
    minIndex = 0
    maxIndex = len(array) - 1
    midIndex = math.floor((minIndex + maxIndex)/2)

    while maxIndex >= minIndex:
        midIndex = math.floor((minIndex + maxIndex)/2)

        if searchFor < array[midIndex]:
            print("lower")
            maxIndex = midIndex - 1

        elif searchFor > array[midIndex]:
            print("higher")
            minIndex = midIndex + 1

        elif searchFor == array[midIndex]:
            print("Found element at index " + str(midIndex))
            break

nInput = input("Search for: ")
binarySearch(testArray, int(nInput))

File: searching/C++/binarySearch.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int binarySearch(vector<int> inArray, int searchFor) {
    int minIndex = 0;
    int maxIndex = inArray.size() - 1;
    int midIndex = floor((minIndex+maxIndex)/2);

    while (maxIndex >= minIndex) {
        midIndex = floor((minIndex+maxIndex)/2);

        if (searchFor < inArray[midIndex]) {
            maxIndex = midIndex - 1;
        }

        else if (searchFor > inArray[midIndex]) {
            minIndex = midIndex + 1;
        }

        else if (searchFor == inArray[midIndex]) {
            return midIndex;
            break;
        }
    }
}

int main() {
    vector<int> testVector = { 1, 3, 4, 5, 9, 11, 18, 22, 31, 42, 69, 100, 101, 102, 450, 999 };
    cout << binarySearch(testVector, 101);
    return 0;
}
